### PR TITLE
Allow dynamic access resource configration

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: pkg/auth/interceptor.go [hunk 1]
```diff
@@ -26,23 +27,48 @@ const (
...
 )
 
+// Config is the configuration for an Interceptor.
+type Config struct {
+	RBACServerAddr string
+
+	// AccessResource is the static resource name to access. This value or GetAccessResource functions must be set.
+	AccessResource string
+	// GetAccessResourceForGRPCRequest is a function to get the resource name from a gRPC method.
+	GetAccessResourceForGRPCRequest func(fullMethod string) string
```
Comment (kkaneda) [inline pkg/auth/interceptor.go:37]: Question/confirmation: Will we add only set `GetAccessResourceForGRPCRequest`? Just curious as the code allows it and it's useful when the server only implements gRPC handlers.

**Review by kkaneda — APPROVED** [review]
LG!

File: pkg/auth/interceptor.go (hunk 1 above)
Comment (Ladicle) [reply pkg/auth/interceptor.go:37]: Yes, this PR supports grpc and http, but I plan to use only grpc for now.

Comment (kenji-cloudnatix) [reply pkg/auth/interceptor.go:37]: Sounds good!

**Pull request merged**